src/auth/digest/Config.cc

```diff
@@ -108,11 +108,14 @@ authDigestNonceEncode(digest_nonce_h * nonce)
     if (nonce->key)
         xfree(nonce->key);
 
-    nonce->key = xcalloc(base64_encode_len(sizeof(digest_nonce_data)), 1);
-    struct base64_encode_ctx ctx;
-    base64_encode_init(&ctx);
-    size_t blen = base64_encode_update(&ctx, reinterpret_cast<char*>(nonce->key), sizeof(digest_nonce_data), reinterpret_cast<const uint8_t*>(&(nonce->noncedata)));
-    blen += base64_encode_final(&ctx, reinterpret_cast<char*>(nonce->key)+blen);
+    SquidMD5_CTX Md5Ctx;
+    HASH H;
+    SquidMD5Init(&Md5Ctx);
+    SquidMD5Update(&Md5Ctx, reinterpret_cast<const uint8_t *>(&nonce->noncedata), sizeof(nonce->noncedata));
+    SquidMD5Final((unsigned char *) H, &Md5Ctx);
```

Comment:
Please avoid C-style casts in his C++ code. Also the H parameter here is an uint8_t[] in the md5.h API.

Reply:
I changed this to a `reinterpret_cast<uint8_t *>`.

The API in [md5.h](https://github.com/squid-cache/squid/blob/SQUID_4_8/include/md5.h#L63) specifies `uint8_t[16]`, but in [md5.c](https://github.com/squid-cache/squid/blob/SQUID_4_8/lib/md5.c#L129-L130) it's `unsigned char[16]`.

```
SQUIDCEXTERN void SquidMD5Final(uint8_t digest[16], struct SquidMD5Context *context);
```

```
void
SquidMD5Final(unsigned char digest[16], struct SquidMD5Context *ctx)
```

Reply:
Hmm, okay, that can be fixed later then.